### PR TITLE
Ip 169

### PR DESCRIPTION
**Pull request by indikaraj**
@priesgo Please could you review

### CODE REVIEW

File: cva-storage/cva-storage-core/src/main/java/org/gel/cva/storage/core/managers/TransactionManager.java [hunk 1]
```diff
@@ -99,14 +100,25 @@ private void update(String id, TransactionStatus status, String errorMessage) th
...
      * @throws CvaException
      */
-    public QueryResult insert(List<ReportedVariant> reportedVariants) throws CvaException {
+    public QueryResult insert(List<ReportedVariant> reportedVariants, String requestSignature) throws CvaException {
         logger.debug("Start inserting transaction ..... ");
+        Query query = new Query(TransactionDBAdaptor.QueryParams.REQUEST_SIGNATURE.key(), requestSignature);
+        List<Transaction> transactionsWithSameSignature = this.adaptor.get(query, null).getResult();
+
+        if (transactionsWithSameSignature != null && !transactionsWithSameSignature.isEmpty()) {
+            boolean condition = transactionsWithSameSignature.stream().allMatch(tr -> tr.getStatus() != TransactionStatus.CANCELLED );
```
Comment (greglever) [inline cva-storage/cva-storage-core/src/main/java/org/gel/cva/storage/core/managers/TransactionManager.java]: do we need the extra space at the end of:
```
!= TransactionStatus.CANCELLED );
```
?

File: cva-storage/cva-storage-core/src/main/java/org/gel/cva/storage/core/managers/TransactionManager.java [hunk 2]
```diff
@@ -99,14 +100,25 @@ private void update(String id, TransactionStatus status, String errorMessage) th
...
+            }
+        }
         Transaction transaction = new Transaction(
                 CvaDateFormatter.getCurrentFormattedDate(true),
                 CvaDateFormatter.getCurrentFormattedDate(false),
                 TransactionStatus.PENDING,
                 reportedVariants,
-                null
+                null,
+                requestSignature
```
Comment (greglever) [inline cva-storage/cva-storage-core/src/main/java/org/gel/cva/storage/core/managers/TransactionManager.java]: can we have a comma after `requestSignature` ? Just so if we add extra parameters it won't look like we're changing two lines ?

File: cva-storage/cva-storage-core/src/main/java/org/gel/cva/storage/core/managers/TransactionManager.java [hunk 3]
```diff
@@ -99,14 +100,25 @@ private void update(String id, TransactionStatus status, String errorMessage) th
...
-    public QueryResult insert(List<ReportedVariant> reportedVariants) throws CvaException {
+    public QueryResult insert(List<ReportedVariant> reportedVariants, String requestSignature) throws CvaException {
         logger.debug("Start inserting transaction ..... ");
+        Query query = new Query(TransactionDBAdaptor.QueryParams.REQUEST_SIGNATURE.key(), requestSignature);
+        List<Transaction> transactionsWithSameSignature = this.adaptor.get(query, null).getResult();
+
+        if (transactionsWithSameSignature != null && !transactionsWithSameSignature.isEmpty()) {
+            boolean condition = transactionsWithSameSignature.stream().allMatch(tr -> tr.getStatus() != TransactionStatus.CANCELLED );
+            if (condition) {
+                logger.error("Transaction insert failure -  Same data intake signature already processed by CVA ");
```
Comment (greglever) [inline cva-storage/cva-storage-core/src/main/java/org/gel/cva/storage/core/managers/TransactionManager.java]: do we need an extra space at the end of the log comment ?

File: cva-storage/cva-storage-mongodb/src/main/java/org/gel/cva/storage/mongodb/adaptors/TransactionMongoDBAdaptor.java [hunk 4]
```diff
@@ -185,5 +187,20 @@ protected Transaction convertAndPopulate(Document document) throws CvaException
...
+                Filters.or(
+                        Filters.eq(TransactionDBAdaptor.QueryParams.STATUS.key(), TransactionStatus.PENDING.name()),
+                        Filters.eq(TransactionDBAdaptor.QueryParams.STATUS.key(), TransactionStatus.BLOCKED.name()),
+                        Filters.eq(TransactionDBAdaptor.QueryParams.STATUS.key(), TransactionStatus.APPLIED.name()),
+                        Filters.eq(TransactionDBAdaptor.QueryParams.STATUS.key(), TransactionStatus.DONE.name()),
+                        Filters.eq(TransactionDBAdaptor.QueryParams.STATUS.key(), TransactionStatus.CANCELLING.name())
+                )
+        );
+        QueryResult<Long> queryResult = super.count(query);
+        return queryResult != null && queryResult.getResult() != null ? queryResult.getResult().size() : 0;
```
Comment (priesgo) [inline cva-storage/cva-storage-mongodb/src/main/java/org/gel/cva/storage/mongodb/adaptors/TransactionMongoDBAdaptor.java]: This is incorrect `queryResult.getResult()` contains a list of Long objects. You will need to fetch the first result `queryResult.first()` and not `queryResult.getResult().size()`

**Pull request merged**